### PR TITLE
feat(config): add ability to define brovider URL

### PR DESCRIPTION
**Pull request by Todmy**
Changes proposed in this pull request:
- add extra variables to functions for configuring hardcoded endpoints/data
- those variables are optional, so it should work the same way for our users

### CODE REVIEW

**Review by samuveth — APPROVED** [review]
utACK

File: src/sign/eip1271.ts [hunk 1]
```diff
@@ -48,8 +48,8 @@ export async function verifyOldVersion(
   return returnValue.toLowerCase() === magicValue.toLowerCase();
 }
 
-export async function verify(address, sig, hash, network = '1') {
-  const provider = getProvider(network);
+export async function verify(address, sig, hash, network = '1', opts) {
+  const provider = getProvider(network, opts);
```
Comment (ChaituVR) [inline src/sign/eip1271.ts:52]: It is not optional? then we should change it in sequencer too? 🤔

File: src/utils.ts [hunk 2]
```diff
@@ -392,15 +398,18 @@ export async function getSpaceController(
...
-  return await getEnsOwner(id, network);
+  return await getEnsOwner(id, network, opts);
 }
 
 export async function getDelegatesBySpace(
   network: string,
   space: string,
-  snapshot = 'latest'
+  snapshot = 'latest',
+  options: any = {}
```
Comment (ChaituVR) [inline src/utils.ts:408]: `opts` or `options` everywhere 😋

File: src/sign/eip1271.ts (hunk 1 above)
Comment (Todmy) [reply src/sign/eip1271.ts:52]: it's optional

File: src/utils.ts [hunk 3]
```diff
@@ -392,15 +398,18 @@ export async function getSpaceController(
...
 ) {
-  if (!delegationSubgraphs[network]) {
+  const subgraphNetwork =
```
Comment (ChaituVR) [inline src/utils.ts:410]: This should be subgraphUrl

File: src/utils.ts [hunk 4]
```diff
@@ -392,15 +398,18 @@ export async function getSpaceController(
...
+    options.subgraphNetwork || SNAPSHOT_SUBGRAPH_URL[network];
+  if (!subgraphNetwork) {
```
Comment (ChaituVR) [inline src/utils.ts:412]:
```suggestion
  const subgraphUrl =
    options.subgraphUrl || SNAPSHOT_SUBGRAPH_URL[network];
  if (!subgraphUrl) {
```

File: src/utils.ts [hunk 5]
```diff
@@ -432,10 +441,7 @@ export async function getDelegatesBySpace(
   while (true) {
     params.delegations.__args.skip = page * PAGE_SIZE;
 
-    const pageResult = await subgraphRequest(
-      delegationSubgraphs[network],
-      params
-    );
+    const pageResult = await subgraphRequest(subgraphNetwork, params);
```
Comment (ChaituVR) [inline src/utils.ts:444]:
```suggestion
    const pageResult = await subgraphRequest(subgraphUrl, params);
```

**Review by ChaituVR — APPROVED** [review]
utAck

**Pull request merged**